src/breadth_first_search/__init__.py

```diff
@@ -0,0 +1,20 @@
+from collections import deque
+
+
+def breadth_first_search(graph, root_node, fn):
+    if not graph:
+        return False
+
+    search_queue = deque()
+    search_queue += graph[root_node]
+    checked = {}
+
+    while search_queue:
+        node = search_queue.popleft()
+        if not checked.get(node):
+            if fn(node):
+                return True
+            else:
+                search_queue += graph[node]
```

Comment:
Maybe it will be better to check `graph[node]` exists and if it doesn't throw humane error instead of "KeyError: 'key'"

Reply:
I guess better just skip it